app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt

```diff
@@ -170,10 +170,18 @@ class ThreadFragment : Fragment() {
 
     private fun displayMessages(messages: List<Message>) {
         Log.i("UI", "Received messages (${messages.size})")
+        leaveIfThreadIsEmpty(messages)
         threadAdapter.notifyAdapter(messages.toMutableList())
         binding.messagesList.scrollToPosition(threadAdapter.lastIndex())
     }
 
+    private fun leaveIfThreadIsEmpty(messages: List<Message>) {
+        if (messages.isEmpty()) {
+            threadViewModel.deleteThread(navigationArgs.threadUid)
+            findNavController().popBackStack()
```

Comment:
If a bottom sheet is open when realm triggers, it might simply close the bottom sheet instead of closing the thread